Sources/Swift/Tools/LoadValidator.swift

```diff
@@ -0,0 +1,90 @@
+@_implementationOnly import _SentryPrivate
+import Darwin
+import Foundation
+import MachO
+
+@objc
+@_spi(Private) public final class LoadValidator: NSObject {
+    // Any class should be fine
+    static let targetClassName = "PrivateSentrySDKOnly"
+    
+    @objc
+    @_spi(Private) public class func validateSDKPresenceIn(_ image: SentryBinaryImageInfo) {
+        internalValidateSDKPresenceIn(image, objcRuntimeWrapper: SentryDefaultObjCRuntimeWrapper.sharedInstance())
+    }
+    
+    /**
+     * This synchronous version is intended to be used in tests.
+     * It uses a higher QoS and wait for the result
+     */
+    @discardableResult class func validateSDKPresenceInSync(_ image: SentryBinaryImageInfo, objcRuntimeWrapper: SentryObjCRuntimeWrapper) -> Bool {
+        var result = false
+        let semaphore = DispatchSemaphore(value: 0)
+        internalValidateSDKPresenceIn(image, objcRuntimeWrapper: objcRuntimeWrapper, qos: .userInitiated) { duplicateFound in
```

Comment:
`m`: I think this can be `qos: .background`. I think it's OK that the log message appears a bit later. This could be a bit expensive and it's better to let it run with low priority in the background.

Reply:
Yes, but many users set `options.debug = true` when setting up Sentry the first time and there are also multiple users keeping this flag on for their debug builds. If this slows them down, they will notice it. I quickly checked for how many images we enqueue an async work item and it's only a few. I think it should be OK to not use a lower priority and simply schedule this to the dispatch queue of the dispatch queue wrapper. It would be good to add this info as a code comment.